docs/configuration.md

```diff
@@ -0,0 +1,167 @@
+
+1. Create a privileged service account:
+
+```console
+$ oc adm new-project wasp
+$ oc create sa -n wasp wasp
+$ oc adm policy add-cluster-role-to-user cluster-admin -z wasp
+$ oc adm policy add-scc-to-user -n wasp privileged -z wasp
+```
+
+2. Configure `Kubelet` to permit swap
+   Create a `KubeletConfiguration` according to the following
+   [example](../manifests/openshift/kubelet-configuration-with-swap.yaml).
+
+3. Create `MachineConfig` to provision swap according to the following [example](../manifests/openshift/machineconfig-add-swap.yaml)
+
+> [!IMPORTANT]
+> In order to have enough swap for the worst case scenario, it must
+> be ensured to have at least as much swap space provisioned as RAM
+> is being over-committed.
+> The amount of swap space to be provisioned on a node must
+> be calculated according to the following formula:
+>
+>     NODE_SWAP_SPACE = NODE_RAM * (MEMORY_OVER_COMMIT_PERCENT / 100% - 1)
+>
+> Example:
+>
+>     NODE_SWAP_SPACE = 16 GB * (150% / 100% - 1)
+>                     = 16 GB * (1.5 - 1)
+>                     = 16 GB * (0.5)
+>                     =  8 GB
+
+Create a `MachineConfig` according to the following
+[example](../manifests/openshift/machineconfig-add-swap.yaml).
+
+4. Deploy `wasp-agent`
+   Create a `DaemonSet` according to the following
+   [example](../manifests/openshift/ds.yaml).
+
+> [!IMPORTANT]
+> The wasp-agent  manages pod eviction when the system is heavily loaded and
+> nodes are at risk. Eviction will be triggered if any of the following conditions occur:
+> 1. High Swap I/O Traffic:
+> * `averageSwapInPerSecond > maxAverageSwapInPagesPerSecond`
+> and<br>
+> `averageSwapOutPerSecond > maxAverageSwapOutPagesPerSecond`<br><br>
+> This condition is triggered when swap-related I/O traffic
+> is excessively high.
+> The default values for `maxAverageSwapInPagesPerSecond` and
+> `maxAverageSwapOutPagesPerSecond` are both set to 1000, averaged 
+> over a 30-second interval by default.
+> 2. High Memory Overcommitment:
+> * `maxMemoryOverCommitmentBytes < SwapUsedBytes - AvailableMemoryBytes`<br><br>
+> This condition is triggered when swap utilization is excessively high.
+> This depends on the `NODE_SWAP_SPACE` setting in the machine configuration from
+> step 3 or when the memory overcommitment ratio is too high.The default value for
+> `maxMemoryOverCommitment` is set to 500Mi.<br><br>
+>
+> `maxAverageSwapInPagesPerSecond`, `maxAverageSwapOutPagesPerSecond`, and 
+> `maxMemoryOverCommitment` can be adjusted by modifying the values of the 
+> `MAX_AVERAGE_SWAP_IN_PAGES_PER_SECOND`, `MAX_AVERAGE_SWAP_OUT_PAGES_PER_SECOND`, 
+> and `MEMORY_OVER_COMMITMENT_THRESHOLD` environment variables in the provided 
+> example, respectively. Additionally, the `AVERAGE_WINDOW_SIZE_SECONDS` environment 
+> variable determines the time frame for calculating the average.
+
+5. Deploy alerting rules according to the following
+   [example](../manifests/openshift/prometheus-rules.yaml).
```

Comment:
Can you please add here:
`oc label namespace wasp "openshift.io/cluster-monitoring=true"`
Since the PrometheusRule is deployed in the wasp namespace.

Reply:
I suggest adding the label directly in the YAML definition to minimize the number of steps users need to perform.

Reply:
I noticed that the old documentation didn't mention creating the namespace. Perhaps we should require users to create the namespace with the appropriate label in place.

Reply:
Done.